src/cogent3/app/composable.py

```diff
@@ -1023,15 +1009,19 @@ def wrapped(klass):
             klass.__slots__ += tuple(slot_attrs)
 
         # register this app
-        __app_registry[get_object_provenance(klass)] = klass.__name__
+        __app_registry[get_object_provenance(klass)] = composable
+
         return klass
 
     return wrapped(klass) if klass else wrapped
 
 
 def is_composable(obj):
     """checks whether obj has been registered by the composable decorator"""
-    return get_object_provenance(obj) in __app_registry
+    return any(
```

Comment:
revert to `__app_registry.get(get_object_provenance(klass), False)`. I'm going to remove the inheritance.